api/app/Http/Resources/UserResource.php

```diff
@@ -32,6 +40,17 @@ public function toArray($request)
         $collection = $collection->merge($personalExperiences);
         $collection = $collection->merge($workExperiences);
 
+        // if poolSkillIds is valid, filter out skills that are not in the id array by stepping through each item in $collection
+        if ($poolSkillIds && count($poolSkillIds) > 0) {
```

Comment:
Careful, you can't for test for null like that in PHP.  However, if you use the `empty` fuction instead of `count` it will handle null safely.
![image](https://user-images.githubusercontent.com/8978655/208926665-ebb184ec-3044-42e6-ab86-c3c5d02d8580.png)

Reply:
ooh, got it, thanks!
https://github.com/GCTC-NTGC/gc-digital-talent/pull/5185/commits/24151228bbfb41ce7bea243955913350978b6313

Reply:
a case where habits from one language slide over to another 😃